### ISSUE TITLE
False positive: 'promise.withresolvers' in 'no-unnecessary-polyfills'

### ISSUE THREAD

**Issue by TimShilov** [sindresorhus/eslint-plugin-unicorn#2404]
The `no-unnecessary-polyfills` rule complains about usage of `promise.withresolvers` package on Node.js 20 even though the native `Promise.withResolvers` is not available. According to MDN, it is only available since Node 22.0.0.

`no-unnecessary-polyfills`

Here's my rule config:
```
"unicorn/no-unnecessary-polyfills": [
    "error",
    { targets: "node 20.15.0" },
],
```

And there's a piece of code that triggers an error.
```typescript
import withResolvers from "promise.withresolvers"; // Use built-in instead. eslint (unicorn/no-unnecessary-polyfills)
```

**Comment by fregante** [comment]
Can you verify that the bug is not actually upstream? Unicorn is using its latest version.

https://github.com/zloirock/core-js/tree/master/packages/core-js-compat

**Comment by zloirock** [comment]
> Can you verify that the bug is not actually upstream?

<img width="1427" alt="Снимок экрана 2024-07-25 в 15 45 29" src="https://github.com/user-attachments/assets/96f56657-f003-42f7-be93-7baa11aa1d39">

**Comment by fregante** [comment]
Thanks @zloirock!
